### PR TITLE
Print case detail

### PR DESCRIPTION
**Pull request by orangejenny**
https://docs.google.com/document/d/18SpmZDc5GvFR2VcYY4KtevIgQQ1aHqDsD0rYMbZLTR0

This code feels atrocious, but it's possible that's just how all multimedia uploader code is.

It also unfortunately reintroduces inline js into `module_view.html`, which I'll deal with at a later date, because it's not straightforward to define `printTemplateUploader` using `initial_page_data`.

Easiest to review all at once.

@NoahCarnahan 
@biyeun can you also look this over? I think you've got the most expertise with the multimedia uploader.

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 5 errors:

* corehq/apps/app_manager/static/app_manager/js/module-view.js, line 42 - 'print_ref' is not defined. (no-undef)
* corehq/apps/hqwebapp/static/hqwebapp/js/key-value-mapping.js, line 32 - 'COMMCAREHQ' is not defined. (no-undef)
* corehq/apps/hqwebapp/static/hqwebapp/js/key-value-mapping.js, line 33 - 'uploaders' is not defined. (no-undef)
* corehq/apps/hqwebapp/static/hqwebapp/js/key-value-mapping.js, line 35 - 'uploaders' is not defined. (no-undef)
* corehq/apps/hqmedia/urls.py, line 38 - E501 line too long (119 > 115 characters)

**Comment by orangejenny** [comment]
@biyeun The main parts that feel sketchy
- In `removePrintTemplate`, I forcibly set `is_matched` to false, because `setObjReference` sets it to true. I couldn't find a canonical way to clear/remove a media file. The closest existing functionality to this PR is the app logo uploader (https://github.com/dimagi/commcare-hq/pull/5249 and https://github.com/dimagi/commcare-hq/pull/5116) but that logic is rather tangled up in the general CommCare settings JavaScript.
- I ended up defining the path (`jr://file/commcare/text/...`) in three places - twice in hqmedia and once in app manager - and would have expected to only define that once.

**Comment by amstone326** [comment]
@orangejenny Would you mind adding a screenshot of what the new UI component looks like?

**Comment by orangejenny** [comment]
@amstone326 Screenshots, with the bottom of the case detail page for context. It looks almost identical to the widget in vellum, with the main difference being that there isn't the option to edit the path.

screen shot 2017-03-28 at 8 16 24 am

screen shot 2017-03-28 at 8 16 47 am

screen shot 2017-03-28 at 8 17 02 am

screen shot 2017-03-28 at 8 17 22 am

File: corehq/apps/app_manager/suite_xml/sections/details.py [hunk 1]
```diff
@@ -72,6 +72,7 @@ def get_section_elements(self):
                                                             detail_type, self.build_profile_id)
                                     r.append(helper.build_case_tile_detail())
                                 else:
+                                    print_template_path = detail.print_template['path'] if detail.print_template else None
```
Comment (dimagimon) [inline corehq/apps/app_manager/suite_xml/sections/details.py]: E501 line too long (122 > 115 characters)

**Comment by amstone326** [comment]
@orangejenny Nice, looks great 👍

**Comment by orangejenny** [comment]
Tests passing.

Would be good to get this reviewed today if possible, so mLabour can do some testing before they go into the field next week.

File: corehq/apps/hqmedia/templates/hqmedia/uploader/preview_html_single.html [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{% load i18n %}
+<p>
+    <a class="btn btn-default existing-media"
+       href="<%= url %>"
```
Comment (NoahCarnahan) [inline corehq/apps/hqmedia/templates/hqmedia/uploader/preview_html_single.html]: What templating system is this for?

**Review by NoahCarnahan — APPROVED** [review]
Just two non-blocking comments. Feel free to self-merge when ready.

File: corehq/apps/app_manager/feature_support.py [hunk 3]
```diff
@@ -121,3 +121,10 @@ def enable_case_list_sort_blanks(self):
         Ability to control where blanks sort for case list sort properties only supported > 2.35
         """
         return self._require_minimum_version('2.35')
+
+    @property
+    def enable_detail_print(self):
+        """
+        Ability to control where blanks sort for case list sort properties only supported > 2.35
```
Comment (orangejenny) [inline corehq/apps/app_manager/feature_support.py]: thanks, fixed

File: corehq/apps/hqmedia/templates/hqmedia/uploader/preview_html_single.html (hunk 2 above)
Comment (orangejenny) [inline corehq/apps/hqmedia/templates/hqmedia/uploader/preview_html_single.html]: underscore: http://underscorejs.org/#template

**Comment by amstone326** [comment]
@orangejenny Hey Jenny, I think something with this wasn't quite right. We tested this for mLabour this morning and got an error with resolving the path to the print template on mobile, and I think I know why. If you look at build 786 of the app (https://www.commcarehq.org/a/mlabour-tz/apps/view/d7cc3f6f1ff4516a9f0ddeebb27e2d65/releases/), the print template that Ali uploaded resulted in the following line being in the suite file:

`<detail print-template="jr://file/commcare/text/module_fbcb3f2f3513ec4e9287811270f40929433b5e3_detail_print.html" id="m2_case_long">`

But if you go to the `/commcare/text` folder within the app files, the only thing that's there is a file called `print-chux4n.html`, which if you open it up, looks like a test print template I sent you, not the template that Ali uploaded. Is it possible something got hard-coded in by mistake?

**Pull request merged**